clcache.py

```diff
@@ -150,7 +150,8 @@ def getManifestHash(compilerBinary, commandLine, sourceFile):
         return getFileHash(sourceFile, additionalData)
 
     @staticmethod
-    def getIncludesContentHash(listOfHeaderHashes):
+    def getIncludesContentHash(listOfIncludes):
+        listOfHeaderHashes = [getFileHash(expandBasedirPlaceholder(fileName, baseDir)) for fileName in listOfIncludes]
         return HashAlgorithm(','.join(listOfHeaderHashes).encode()).hexdigest()
```

Comment:
At this point, I guess one could ask why

1. The code needs a list at all, only to join it again in the next line and
2. Why we're hashing a list of hashes here, instead of hashing the file contents directly (changing this probably means that the hash sums change).

I.e. I suspect something like

```python
hasher = HashAlgorithm()
for fileName in listOfIncludes:
    with open(expandBasedirPlaceholder(fileName, baseDir), 'rb') as f:
        hasher.update(f.read())
return hasher.hexdigest()
```

...might be a good idea. However, all this is out of scope for this particular commit, of course (I like how you're taking small steps!).